pkg/parser/parser.go

```diff
@@ -556,11 +556,12 @@ func (p *parser) assertArgTypes(decl *FuncDeclStmt, args []Node) {
 		p.appendError("'" + funcName + "' takes " + quantify(len(decl.Params), "argument") + ", found " + strconv.Itoa(len(args)))
 		return
 	}
-	for i := range args {
+	for i, arg := range args {
 		paramType := decl.Params[i].Type()
-		argType := args[i].Type()
+		argType := arg.Type()
 		if !paramType.Accepts(argType) && !paramType.Matches(argType) {
-			p.appendError("'" + funcName + "' takes " + ordinalize(i+1) + " argument of type '" + paramType.String() + "', found '" + argType.String() + "'")
+			msg := fmt.Sprintf("'%s' takes %s argument of type '%s', found '%s'", funcName, ordinalize(i+1), paramType.String(), argType.String())
+			p.appendErrorForToken(msg, arg.Token())
```

Comment:
in my local proof of concept, I went a bit further and also added the token to the two other errors in this function:

```
                        if !paramType.Accepts(argType) && !paramType.Matches(argType) {
-                               p.appendError("'" + funcName + "' takes variadic arguments of type '" + paramType.String() + "', found '" + argType.String() + "'")
+                               p.appendErrorForToken("'"+funcName+"' takes variadic arguments of type '"+paramType.String()+"', found '"+argType.String()+"'", arg.Tok())
                        }

```

and

```
        if len(decl.Params) != len(args) {
-               p.appendError("'" + funcName + "' takes " + quantify(len(decl.Params), "argument") + ", found " + strconv.Itoa(len(args)))
+               tok := p.cur
+               if len(args) > len(decl.Params) {
+                       tok = args[len(decl.Params)].Tok()
+               }
+               p.appendErrorForToken("'"+funcName+"' takes "+quantify(len(decl.Params), "argument")+", found "+strconv.Itoa(len(args)), tok)
                return
        }

```

Reply:
all addressed. thank you!
(I had a mental note of doing this and then I forgot. busy mind)